sessiontxn/interface.go

```diff
@@ -137,8 +137,14 @@ type TxnContextProvider interface {
 	// OnStmtStart is the hook that should be called when a new statement started
 	OnStmtStart(ctx context.Context, node ast.StmtNode) error
 	// OnHandlePessimisticStmtStart is the hook that should be called when starts handling a pessimistic DML or
-	// a pessimistic select-for-update statements.
+	// a pessimistic select-for-update statement.
 	OnHandlePessimisticStmtStart(ctx context.Context) error
+	// OnPessimisticLockError is the hook that should be called when lock error happens during a pessimistic DML or
+	// select-for-update statement.
+	OnPessimisticLockError(ctx context.Context, lockErr error) error
```

Comment:
`OnStmtErrorForNextAction` has some logic about handling `StmtErrAfterPessimisticLock`, could we merge these logic together so we could avoid adding a new interface.

Reply:
That makes sense and looks safe to use. I didn't notice that.